lib/src/review_service/review_service.extensions.dart

```diff
@@ -0,0 +1,34 @@
+import 'package:review_service/src/review_service/review_service.dart';
+import 'package:review_service/src/review_service/review_settings.dart';
+
+/// Extensions of [IReviewService].
+extension ReviewServiceExtensions<TReviewSettings extends ReviewSettings> on ReviewService<TReviewSettings> {
+
+  /// Tracks that the application was launched.
+  Future<void> trackApplicationLaunched() async {
+    await updateReviewSettings((reviewSettings) {
+      return reviewSettings.firstApplicationLaunch == null
+          ? reviewSettings.copyWith(
```

Comment:
Does this work with custom settings types? I think that it might not because `copyWith` will not return the correct (derived) type.

Reply:
Do you mean the application launched or the update review settings method ?

Reply:
If you meant the application launched tracking I retested everything and it all works

Reply:
I tested it and it works, althought you gotta specify the customReviewSettings everywhere

Reply:
Hmm what do you mean by that? Can you write an automated test that ensures that custom property values aren't lost when using a custom `ReviewSettings` type and the `trackApplicationLaunched` method?